clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -2064,6 +2064,9 @@ llvm::Value *CodeGenFunction::EmitToMemory(llvm::Value *Value, QualType Ty) {
 
   if (Ty->isExtVectorBoolType()) {
```

Comment:
you replaced a bunch of `isExtVectorBoolType()` with `isPackedVectorBoolType`.  We are only doing HLSL modifications on the `isExtVectorBoolType()`. But `isExtVectorBoolType` doesn't mean the vector is not packed. Is the Zero extend and truncation to get them into a form that they will unpack?

Reply:
My intention was to replace calls to 'isExtVectorBoolType' and 'isPackedVectorBoolType' anywhere we want an hlsl boolean vector to follow the normal handling path for vectors; Hopefully reviews will verify I got this right.
Here we can't follow the normal vector path because it returns the value unchanged, and we need to convert a vec of i1s to a vec of i32s, which is why we zero extend here. The normally "boolean vector packing" does something different.